src/model_io/urdf/src/URDFModelExport.cpp

```diff
@@ -0,0 +1,295 @@
+    {
+        xmlNewProp(joint_xml, BAD_CAST "type", BAD_CAST "fixed");
+    }
+    else if (dynamic_cast<const RevoluteJoint*>(joint))
+    {
+        if(joint->hasPosLimits())
+        {
+            xmlNewProp(joint_xml, BAD_CAST "type", BAD_CAST "revolute");
+        }
+        else
+        {
+            xmlNewProp(joint_xml, BAD_CAST "type", BAD_CAST "continuous");
+        }
+    }
+    else if (dynamic_cast<const PrismaticJoint*>(joint))
+    {
+        xmlNewProp(joint_xml, BAD_CAST "type", BAD_CAST "prismatic");
+    }
+    else
+    {
+        std::cerr << "[ERROR] URDFModelExport: Impossible to convert joint of type "
+                  <<  typeid(joint).name() << " to a URDF joint " << std::endl;
+        return false;
+    }
+
+    // origin
+    exportTransform(joint->getRestTransform(parentLink->getIndex(), childLink->getIndex()), joint_xml);
+
+    if (joint->getNrOfDOFs() != 0)
+    {
+        Axis axis;
+
+        // Check that the axis of the joint is supported by URDF
+        if (dynamic_cast<const RevoluteJoint*>(joint))
+        {
+            const RevoluteJoint* revJoint = dynamic_cast<const RevoluteJoint*>(joint);
+            axis = revJoint->getAxis(childLink->getIndex());
+        }
+        else if (dynamic_cast<const PrismaticJoint*>(joint))
+        {
+            const PrismaticJoint* prismJoint = dynamic_cast<const PrismaticJoint*>(joint);
+            axis = prismJoint->getAxis(childLink->getIndex());
+        }
+        else
+        {
+            std::cerr << "[ERROR] URDFModelExport: Impossible to convert joint of type "
+                      <<  typeid(joint).name() << " to a URDF joint " << std::endl;
+            return false;
+        }
+
+        // Check that the axis is URDF-compatible
+        if (toEigen(axis.getOrigin()).norm() > 1e-7)
+        {
+            std::cerr << "[ERROR] URDFModelExport: Impossible to convert joint "
+                      <<   model.getJointName(joint->getIndex()) << " to a URDF joint without moving the link frame of link "
+                      << model.getLinkName(childLink->getIndex()) << " , because its origin is "
+                      << axis.getOrigin().toString()  << std::endl;
+            return false;
+        }
+
+        // axis
+        xmlNodePtr axis_xml = xmlNewChild(joint_xml, NULL, BAD_CAST "axis", NULL);
+        std::string bufStr;
+        ok = ok && vectorToString(axis.getDirection(), bufStr);
+        xmlNewProp(axis_xml, BAD_CAST "xyz", BAD_CAST bufStr.c_str());
+    }
+
+    // parent
+    xmlNodePtr parent_xml = xmlNewChild(joint_xml, NULL, BAD_CAST "parent", NULL);
+    xmlNewProp(parent_xml, BAD_CAST "link", BAD_CAST model.getLinkName(parentLink->getIndex()).c_str());
+
+    // child
+    xmlNodePtr child_xml = xmlNewChild(joint_xml, NULL, BAD_CAST "child", NULL);
+    xmlNewProp(child_xml, BAD_CAST "link", BAD_CAST model.getLinkName(childLink->getIndex()).c_str());
+
+    // TODO(traversaro) : handle joint limits and friction
```

Comment:
This should be moved also in the documentation.

Reply:
I agree this is not mentioned in the description of the pr

Reply:
I think this is still pending